### DISCUSSION TITLE
Can I link to a section (navigation index) from a markdown file?

### DISCUSSION THREAD

**Discussion by emign** [squidfunk/mkdocs-material#4186] [Q&A]
Hi,
I have the following folder structure:

```
Samenpflanzen als Lebewesen/
├─ index.md
├─ Samenpflanzen als.md
Der Mensch als Lebewesen/
├─ index.md
├─ Stoffwechsel/
│  ├─ index.md
├─ Information - Aufnahme, Verarbeitung und Reaktion.md
Ökosystem Grünland.md
```
The contents of `Ökosystem Grünland.md` are:
```
---
---
[[Samenpflanzen als]]

[[Samenpflanzen als Lebewesen]]

[[Der Mensch als Lebewesen]]

[[Stoffwechsel]]

[[Information - Aufnahme, Verarbeitung und Reaktion]]
```

You can open the resulting mkdocs generated html file here: [https://unterricht.emig.me/Unterricht/F%C3%A4cher/Biologie/NuT/%C3%96kosystem%20Gr%C3%BCnland/](https://unterricht.emig.me/Unterricht/F%C3%A4cher/Biologie/NuT/%C3%96kosystem%20Gr%C3%BCnland/)

As you can see, the links to `Samenpflanzen als.md` and `Information - Aufnahme, Verarbeitung und Reaktion.md` are working perfectly fine. But the links to the sections with the `index.md files do not work.

For example the `Stoffwechsel` link is generated as : `<a href="../../../../Stoffwechsel" title="Stoffwechsel">Stoffwechsel</a>`
but it should be: `<a href="../Der%20Mensch%20als%20Lebewesen/Stoffwechsel/">Stoffwechsel</a>`

Please note, that the link in the navigation bar on the left actually IS generated correctly 

How can I correctly link to the navigation index?

**Comment by squidfunk** [comment]
Just link to the respective `index.md` file. Note that all Markdown links need to be relative from the linking document.

**Reply by emign** [reply]
I think this will be my workaround for now.

**Reply by squidfunk** [reply]
You should always link to `*.md` files, not to the resulting URLs. It's not a workaround.

**Reply by squidfunk** [reply]
... ah you're using some wiki-like extension. Might be worth asking the maintainers of the extension.

**Reply by emign** [reply]
Yes. The files are generated in obsidian

**Reply by emign** [reply]
I still think it's an mkdocs/material issue because the nav links on the left are created correctly from the same source

**Reply by squidfunk** [reply]
I'd recommend trying another theme (e.g. `mkdocs`). My intuition is that you'll end up with the same problem which would indicate that it's not a Material for MkDocs issue, but rather a Python Markdown or other extension issue.

**Reply by emign** [reply]
Only material has the _navigation.indexes _ extension (which correctly generates the links). It's just the page internal links which don't work. And not because they don't reference the correct base-file/folder but for them **../../../../../** is prepended to the link.

**Reply by squidfunk** [reply]
> Please note, that the link in the navigation bar on the left actually IS generated correctly

`navigation.indexes` only impacts the navigation bar.

**Reply by emign** [reply]
> > Please note, that the link in the navigation bar on the left actually IS generated correctly
> 
> 
> 
> `navigation.indexes` only impacts the navigation bar.

Yes. But both (Navbar links and page links) are generated from the same (md) source files. Still the navbar links work, the page links break in the process. 

The navbar links keep working ok any subpage too.
